include/denovo_discovery/candidate_region.h

```diff
@@ -87,26 +87,37 @@ class CandidateRegion {
     Fastaq generate_fasta_for_denovo_paths();
 };
 
-std::vector<Interval> identify_low_coverage_intervals(
-    const std::vector<uint32_t>& covg_at_each_position,
-    const uint32_t& min_required_covg = 2, const uint32_t& min_length = 1,
-    const uint32_t& max_length = 20);
+using CandidateRegions
+= std::unordered_map<CandidateRegionIdentifier, CandidateRegion>;
+using ReadId = uint32_t;
+using PileupConstructionMap = std::map<ReadId,
+    std::vector<std::pair<CandidateRegion*, const ReadCoordinate*>>>;
 
-using CandidateRegions = std::unordered_map<CandidateRegionIdentifier, CandidateRegion>;
+class Discover {
+private:
+    const uint32_t min_required_covg;
+    const uint32_t min_candidate_len;
+    const uint32_t max_candidate_len;
+    const uint16_t candidate_padding;
+    const uint32_t merge_dist;
+public:
 
-CandidateRegions find_candidate_regions_for_pan_node(
-    const TmpPanNode& pangraph_node_components,
-    const uint_least16_t& candidate_region_interval_padding = 0);
+    Discover(uint32_t, uint32_t, uint32_t, uint16_t, uint32_t);
```

Comment:
Very minor: I have a preference to have the parameter names in the header file. Although it is not required from the compilation POV, the idea is that the header file presents the interface of a module/class, and hides the implementation from the user (which usually has no need to see the details), but when the parameters names are hidden in the header, we have to forcibly go to the implementation/cpp file to actually see what each parameter is. This is minor because this is just a few clicks away, and CLion I think takes care of this with inspections

Reply:
Ah ok. I also have the same preference, but I had noticed a lot of methods in the code base *didn't* name them so I was just trying to follow convention. However, I agree it is better to name them. Will fix this now.